src/versionbits.cpp

```diff
@@ -136,6 +136,27 @@ class VersionBitsConditionChecker : public AbstractThresholdConditionChecker {
 
 }
 
+bool IsVersionBitRejected(const Consensus::Params& params, Consensus::DeploymentPos pos){
+
+    bool isRejected = false;
+
+    std::vector<std::string>& versionBitVotes = mapMultiArgs["-rejectversionbit"];
+
+    int bitTest = params.vDeployments[pos].bit;
+
+     BOOST_FOREACH(std::string rejectedBit, versionBitVotes) {
+       int rBit =  stoi(rejectedBit);
+
+       if(rBit == bitTest) {
```

Comment:
From what I understand reading the code we are looping through all the deployments defined in chainparams.cpp which is good. we don't need to test if the number is out of range or if the bit exists.

Are there any version bits which we don't want to be disabled in this way? Is there any ill effect if the network turns off any of the other version bits?

https://github.com/aguycalled/navcoin-core/blob/bc765f242c2a59f684068eeef323fff376f34e5a/src/chainparams.cpp#L126

I assume it will have no effect, but wouldn't mind a second opinion @aguycalled

Reply:
Im not to sure @aguycalled ?

Reply:
from what i read in the code, he's looping through the different values given to the key "rejectversionbit" in the config file. the only optimisation i can see here would be to directly return true whenever a match is found, to save from going through the next iterations. the necessary bit masks are set independent from this check, so i don't consider any risk